### PR TITLE
Add the ability to add extra fixtures

### PR DESCRIPTION
**Pull request by cybercapsicum**
This Pull Request adds additional configuration options to the DefectDojo Helm chart by mounting extra fixtures as a ConfigMap volume to the initializer job. The `extra_*.json` files from this volume are copied to the `/app/dojo/fixtures/` directory to be run by the existing [`entrypoint-initializer.sh`](https://github.com/DefectDojo/django-DefectDojo/blob/33d387e7682a410b336f528078a0ade38c42c00e/docker/entrypoint-initializer.sh#L100).

### CODE REVIEW

**Comment by cybercapsicum** [comment]
The current script will only import fixtures if the admin user doesn't exist ([ref](https://github.com/DefectDojo/django-DefectDojo/blob/33d387e7682a410b336f528078a0ade38c42c00e/docker/entrypoint-initializer.sh#L60)), which is only the case on a fresh installation.

**Comment by cybercapsicum** [comment]
In my test environment, the extra fixtures were correctly imported.

File: helm/defectdojo/templates/extra-fixtures.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    {{- $models := dict }}
+    {{- $output := list }}
+    {{- range $j, $data := $fixture.data }}
+    {{- if empty $data.model }}
+    {{- printf "'model' key missing from extraFixture %d, data item %d" $i $j | fail }}
+    {{- end }}
+    {{- if empty $data.fields }}
+    {{- printf "'fields' key missing from extraFixture %d, data item %d" $i $j | fail }}
+    {{- end }}
+    {{- $_ := set $models $data.model (get $models $data.model | add1) }}
```
Comment (cybercapsicum) [inline helm/defectdojo/templates/extra-fixtures.yaml:23]: I would especially appreciate feedback on this line, which increments the `pk` value for every additional item in the fixture with that same `model`. Is generating the `pk` this way at all problematic? I am not very familiar with the project, so I would appreciate some help here.

File: helm/defectdojo/templates/extra-fixtures.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    helm.sh/chart: {{ include "defectdojo.chart" . }}
+data:
+  {{- range $i, $fixture := .Values.extraFixtures }}
+  {{- printf "extra_%03d.json" $i | nindent 2 }}: |
+    {{- $models := dict }}
+    {{- $output := list }}
+    {{- range $j, $data := $fixture.data }}
+    {{- if empty $data.model }}
+    {{- printf "'model' key missing from extraFixture %d, data item %d" $i $j | fail }}
```
Comment (cybercapsicum) [inline helm/defectdojo/templates/extra-fixtures.yaml:18]: I decided it was important to fail helm installations if extra fixtures are missing the required fields. Is this important?

**Review by cybercapsicum — COMMENTED** [review]
Some lines I would appreciate feedback on

File: helm/defectdojo/values.yaml [hunk 3]
```diff
@@ -495,6 +495,16 @@ redis:
...
 
+# Extra fixtures can be mounted to the initializer job by adding an item to the extraFixtures list:
+# NOTE  Each fixture must have a 'data' section, which must include a 'model' string and a 'fields' map
+# NOTE  These fixtures are only applied to an uninitialized database
+# extraFixtures:
+# - data:
+#   - model: dojo.system_settings
+#     fields:
+#       enable_deduplication: true
+extraFixtures: []
```
Comment (dsever) [inline helm/defectdojo/values.yaml:506]: Can you provide example how extraFixuters should look like?

Comment (dsever) [reply helm/defectdojo/values.yaml:506]: Is this the usecase? If it is, you should use api to bootstrap your app.

Comment (cybercapsicum) [reply helm/defectdojo/values.yaml:506]: It's the simplest usecase I could think of, but users could use this to pre-populate their databases with anything from additional test types to custom surveys

Comment (cybercapsicum) [reply helm/defectdojo/values.yaml:506]: The API would be better for most post-install configurations like this. I may go that direction for what I initially made this for, but this feature is still potentially useful. Whether it's useful enough to maintain is up to the you all

**Comment by cybercapsicum** [comment]
Just updated to create a `Secret` rather than a `ConfigMap`, so users aren't limited by the sensitivity of data. I also added the ability to set the `pk` so users have more control. If the `pk` isn't set, it will increment like normal.

**Comment by dsever** [comment]
I would like to close this PR without merging it, to keep the helm simple as possible. @alles-klar @cybercapsicum do you agree?

**Comment by alles-klar** [comment]
I agree. This feature will be hard to maintain and we have the API for most use cases. 

Anyway, thanks for your work @cybercapsicum!

**Comment by dsever** [comment]
Thanks @cybercapsicum for PR, Closing PR for now

**Pull request closed without merging**